agent/config/builder.go

```diff
@@ -436,6 +436,9 @@ func (b *builder) build() (rt RuntimeConfig, err error) {
 	serverPort := b.portVal("ports.server", c.Ports.Server)
 	grpcPort := b.portVal("ports.grpc", c.Ports.GRPC)
 	grpcTlsPort := b.portVal("ports.grpc_tls", c.Ports.GRPCTLS)
+	if grpcTlsPort == -1 && boolVal(c.ServerMode) { // default gRPC TLS port for servers is 8503
```

Comment:
If we check after doing the `b.portVal()` call, then users will have no way to disable the port manually.

We should only set the default if `c.Ports.GRPCTLS == nil`, because that will mean that the field was never populated with a value.